runelite-client/src/main/java/net/runelite/client/plugins/deathindicator/DeathIndicatorPlugin.java

```diff
@@ -0,0 +1,267 @@
+		{
+			return;
+		}
+
+		Actor actor = event.getActor();
+
+		if (actor instanceof NPC || actor != client.getLocalPlayer())
+		{
+			return;
+		}
+
+		hasDied = true;
+		playerHasRespawned = false;
+
+		deathLocation = actor.getWorldLocation();
+		deathWorld = client.getWorld();
+
+		if (config.showDeathHintArrow())
+		{
+			client.setHintArrow(deathLocation);
+		}
+
+		if (config.showDeathOnWorldMap())
+		{
+			addMapPoint(deathLocation);
+		}
+
+		if (deathTimer != null)
+		{
+			infoBoxManager.removeInfoBox(deathTimer);
+		}
+
+		if (config.showDeathInfoBox())
+		{
+			deathTimer = new Timer(60, ChronoUnit.MINUTES, BONES, plugin);
+			deathTimer.setTooltip("Died on world: " + deathWorld);
+			infoBoxManager.addInfoBox(deathTimer);
+		}
+	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		if (event.getMessage().equals("Oh dear, you are dead!"))
+		{
+			playerHasRespawned = true;
+		}
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		if (!hasDied)
+		{
+			return;
+		}
+
+		if (!playerHasRespawned)
+		{
+			return;
+		}
+
+		currentWorld = client.getWorld();
+
+		if (currentWorld != deathWorld)
+		{
+			return;
+		}
+
+		Player player = client.getLocalPlayer();
+
+		currentLocation = player.getWorldLocation();
+
+		if ((currentLocation.getX() == deathLocation.getX())
+				&& (currentLocation.getY() == deathLocation.getY())
+				&& (currentLocation.getPlane() == deathLocation.getPlane())
+				&& (currentLocation.getRegionID() == deathLocation.getRegionID()))
+		{
+			client.clearHintArrow();
+			infoBoxManager.removeInfoBox(deathTimer);
+			worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+			worldMapPointSet = false;
+			deathMarkerReset = true;
+		}
+
+		if (deathMarkerReset)
+		{
+			hasDied = false;
+			playerHasRespawned = true;
+			deathMarkerReset = false;
+		}
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (event.getGroup().equals("deathindicator") && !config.showDeathHintArrow())
+		{
+			client.clearHintArrow();
+		}
+
+		if (event.getGroup().equals("deathindicator") && !config.showDeathInfoBox())
+		{
+			infoBoxManager.removeInfoBox(deathTimer);
+		}
+
+		if (event.getGroup().equals("deathindicator") && !config.showDeathOnWorldMap())
+		{
+			worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+			worldMapPointSet = false;
+		}
+	}
+
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged event)
+	{
+		if (!hasDied)
+		{
+			return;
+		}
+
+		currentWorld = client.getWorld();
+		if (currentWorld == deathWorld)
+		{
+			if (config.showDeathHintArrow())
+			{
+				client.setHintArrow(deathLocation);
+			}
+
+			if (config.showDeathOnWorldMap())
+			{
+				addMapPoint(deathLocation);
+			}
+		}
+		else
+		{
+			if (worldMapPointSet)
+			{
+				worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+				worldMapPointSet = false;
+			}
+		}
+	}
+
+	private void addMapPoint(WorldPoint point)
```

Comment:
This can be in lined to be cleaner. Always remove the point before adding it and if it isn't already there it will remove nothing and be fine.
This also eliminates the Boolean

Reply:
Done, Thank you!